src/ui/components/RecoverySeedPhraseModule/RecoverySeedPhraseModule.tsx

```diff
@@ -0,0 +1,438 @@
+    };
+  }, [lastLockTime]);
+
+  useEffect(() => {
+    if (!lastLockTime) return;
+
+    const lockTime = LOCK_TIME - getLockDuration();
+
+    if (lockTime <= 0) {
+      setLockFailAttempt(false);
+      return;
+    }
+
+    setLockFailAttempt(true);
+    setFailAttempt(MAX_ATTEMPT_FAIL);
+
+    const timerId = setTimeout(() => {
+      setLockFailAttempt(false);
+    }, lockTime);
+
+    return () => {
+      clearTimeout(timerId);
+    };
+  }, [lastLockTime]);
+
+  const handleClearState = () => {
+    setSeedPhraseInfo([
+      {
+        value: "",
+        suggestions: [],
+      },
+    ]);
+    setLastFocusIndex(null);
+    setAlertIsOpen(false);
+    seedPhraseRef.current?.focusInputByIndex(0);
+  };
+
+  useImperativeHandle(ref, () => ({
+    clearState: handleClearState,
+  }));
+
+  const renderSuggestSeedPhrase = (inputWord: string) => {
+    inputWord = inputWord.toLowerCase().trim();
+
+    if (!inputWord) {
+      return [];
+    }
+
+    const suggestionWords: string[] = [];
+    for (let index = 0; index < SELECT_WORD_LIST.length; index++) {
+      const word = SELECT_WORD_LIST[index];
+
+      if (word.startsWith(inputWord)) {
+        suggestionWords.push(word);
+      }
+
+      if (suggestionWords.length >= SUGGEST_SEED_PHRASE_LENGTH) {
+        break;
+      }
+    }
+
+    return suggestionWords;
+  };
+
+  const handleUpdateSeedPhrase = (value: string, index: number) => {
+    const currentValue = [...seedPhraseInfo];
+
+    const suggestions = renderSuggestSeedPhrase(value);
+
+    currentValue[index] = {
+      value,
+      suggestions: suggestions,
+    };
+
+    if (!value && index === currentValue.length - 2) {
+      currentValue.splice(currentValue.length - 2, 1);
+    } else if (
+      index + 1 === currentValue.length &&
+      index + 1 < SEED_PHRASE_LENGTH
+    ) {
+      currentValue.push({
+        value: "",
+        suggestions: [],
+      });
+    }
+
+    setSeedPhraseInfo(currentValue);
+
+    return currentValue;
+  };
+
+  const addSeedPhraseSelected = (word: string) => {
+    if (lastFocusIndex === null) return;
+
+    const newValue = handleUpdateSeedPhrase(word, lastFocusIndex);
+
+    if (
+      lastFocusIndex !== SEED_PHRASE_LENGTH - 1 &&
+      filledSeedPhrase.length !== SEED_PHRASE_LENGTH
+    ) {
+      seedPhraseRef.current?.focusInputByIndex(newValue.length - 1);
+    }
+  };
+
+  const handleContinue = async () => {
+    try {
+      const seedPhraseText = seedPhrase.join(" ");
+      const verifyResult = await Agent.agent.isMnemonicValid(seedPhraseText);
+
+      if (!verifyResult) {
+        throw new Error(INVALID_SEED_PHRASE);
+      }
+
+      dispatch(
+        setSeedPhraseCache({
+          seedPhrase: seedPhraseText,
+          bran: "",
+        })
+      );
+      SecureStorage.delete(KeyStoreKeys.RECOVERY_WALLET_LAST_FAIL_ATTEMPT_TIME);
+      setFailAttempt(0);
+
+      onVerifySuccess();
+    } catch (e) {
+      setFailAttempt((value) => value + 1);
```

Comment:
Also mentioned in the other PR is we shouldn't tackle this - it was a future consideration in the ticket, not an acceptance criteria.

This solution won't work since the user can just restart the app to clear the fail attempt - it all needs to be in the core, and done more carefully to also capture login attempts. This is a future core ticket so I would remove any checks for failed attempts from this PR and handle it later. We can use this as a reference later

Reply:
@iFergal 
I removed check fail attempt. Please help me check it.

https://github.com/cardano-foundation/cf-identity-wallet/assets/162310763/2964bab6-68ac-442a-82c4-ade90c4ab021

cc: @sdisalvo-crd